### PR TITLE
feat(sdk): DSPX-2418 add attribute discovery methods

### PR DESCRIPTION
**Pull request by marythought**
## Summary

- Add five discovery methods in `lib/src/policy/discovery.ts`, exported from the package root:
  * `listAttributes()`, 
  * `validateAttributes()`, 
  * `attributeExists()`,
  * `attributeValueExists()`, 
  * `getEntityAttributes()` -- REMOVED (not applicable)
  
- Add `AttributeNotFoundError extends TdfError`, exported from the package root so callers can
  distinguish "not found" from other errors
- Add unit tests in `discovery.spec.ts` covering all methods and edge cases
- docs update: https://github.com/opentdf/docs/pull/186

## Rationale

Addresses GitHub discussion developer feedback. These methods provide ergonomic wrappers over the
lower-level `PlatformClient` that:

- Auto-paginate `listAttributes()` with a 1,000-page safety cap to prevent unbounded memory
  growth from a misbehaving server
- Validate FQN format and existence before `createZTDF()` so failures happen at encryption time
  with a clear message rather than at decryption time with a cryptic "resource not found"
- Match entity ID in the `GetEntitlements` response to prevent silent wrong-entity attribute
  returns
- Cap `validateAttributes()` at 250 FQNs client-side to match the server-side limit
- Restrict FQN strings to safe URL characters to prevent XSS via FQNs in error messages

Matches the API shape and security properties of the Go SDK implementation: opentdf/platform#3082.

## Test plan

- [ ] `npm test` in `lib/` — all tests pass, no regressions

### CODE REVIEW

File: lib/src/policy/discovery.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+  const vals = resp.attribute?.values ?? [];
+  if (vals.length === 0) {
+    // Dynamic attribute — any value is permitted.
+    return;
+  }
+
+  const match = vals.some((v) => v.value.toLowerCase() === value.toLowerCase());
+  if (!match) {
+    throw new AttributeNotFoundError('attribute value not permitted for this attribute');
```
Comment (eugenioenko) [inline lib/src/policy/discovery.ts]: **nit** : 

```value "${value}" is not permitted for attribute ${attributeFqn}```

File: lib/src/policy/discovery.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+ * @example
+ * ```ts
+ * import type { Entity } from '@opentdf/sdk';
+ *
+ * const entity: Entity = { id: 'e1', emailAddress: 'alice@example.com' };
+ * const fqns = await getEntityAttributes(platformUrl, authProvider, entity);
+ * console.log("alice's entitlements:", fqns);
+ * ```
+ */
+export async function getEntityAttributes(
```
[the remark this answers is not in the document]
Comment (marythought) [reply lib/src/policy/discovery.ts]: That makes sense that it would be useful in the Go/Java versions for limited entities (a service user), but not applicable to the web SDK. In general we want feature parity across languages, but I think that adds confusion/risk here and support removing it.

File: lib/src/policy/discovery.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+ * await validateAttributes(platformUrl, authProvider, [
+ *   'https://opentdf.io/attr/department/value/marketing',
+ * ]);
+ * // Safe to encrypt — all attributes confirmed present
+ * ```
+ */
+export async function validateAttributes(
+  platformUrl: string,
+  authProvider: AuthProvider,
+  fqns: string[]
```
[the remark this answers is not in the document]
Comment (eugenioenko) [reply lib/src/policy/discovery.ts]: @marythought  is this a valid comment or gemini overthinking?

[the remark this answers is not in the document]
Comment (marythought) [reply lib/src/policy/discovery.ts]: @eugenioenko sorry I missed this! Yes, this is gemini being overcautious -- in response to this comment I added a comment on line 28 of discovery.ts: `// Restricts to safe URL characters to prevent XSS via FQNs in error messages — explicitly documenting the intent.`

and added two tests in the validateAttributes suite of discovery.spec.ts:

- throws ConfigurationError for FQN with HTML tags (XSS prevention) — tests that <script> in an FQN is rejected
- throws ConfigurationError for FQN with special characters — tests that &param=1 is rejected
